src/UI/Component/Floatable/Factory.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/* Copyright (c) 2017 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Floatable;
+
+use ILIAS\UI\Component\Component;
+/**
+ * Floatable factory
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       Instructional Overlays keep didactical instructions available to users
+	 *       while they carry out workflows presented by this instruction that spread
+	 *       over several screens. Instructional information and actions are made
+	 *       available at the very screen the user is working on.
+	 *   composition: >
+	 *       The Instructional Overlay is comprised of a title and a glyphed
+	 *       close-button, an assignment description (as shown in the info-tab of e.g. the
+	 *       exercise) namely textual work instructions, accompanying files as links or
+	 *       inline media, schedule information if applicable and some
+	 *       submission-Button to carry out action the single most important action of
+	 *       the workflow.
```

Comment:
What happens if the "single most important action" cannot be carried out (e.g. some config is missing)? Can the floatable be modified by the screen I am watching?